### ISSUE TITLE
padding-line-between-statements fixes code wrong

### ISSUE THREAD

**Issue by hipstersmoothie** [typescript-eslint/typescript-eslint#721]
**What code were you trying to parse?**

The newline at the end is important

```ts
const digits: number[] = [];
const digit = digits.pop();
if (digit) {
  console.log(digit);
}

```

```json
{
  "rules": {
    'padding-line-between-statements': [
      'warn',
      {
        blankLine: 'always',
        prev: '*',
        next: ['return', 'block', 'block-like']
      }
    ]
  }
}
```

**What did you expect to happen?**

```ts
const digits: number[] = [];
const digit = digits.pop();

if (digit) {
  console.log(digit);
}

```

**What actually happened?**

In some cases it add a whole lot more than 2 extra newlines. and in some cases it even mangles the code

```ts
const digits: number[] = [];
const digit = digits.pop();


if (digit) {
  console.log(digit);
}

```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.9.0` |
| `TypeScript`                | `3.5.2` |
| `ESLint`                    | `5.16.0` |
| `node`                      | `10.15.0` |
| `npm`                       | `6.9.0` |

**Comment by platinumazure** [comment]
Hi @hipstersmoothie, thanks for the issue.

I'm wondering if this is because `block` and `block-like` statement types are actually overlapping, and so the rule is catching the same statement pair twice. Can you try using just `block-like` (which should include everything that counts as a `block` as well) and see if the autofix is more reasonable in that case?

**Comment by bradzacher** [comment]
As @platinumazure mentioned - this is an issue with a core rule, and the problem exists without typescript code (i.e. drop the `: number[]`, and the problem still exists)

[repl](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHBhZGRpbmctbGluZS1iZXR3ZWVuLXN0YXRlbWVudHM6IFsnZXJyb3InLCB7XG4gICAgICAgIGJsYW5rTGluZTogJ2Fsd2F5cycsXG4gICAgICAgIHByZXY6ICcqJyxcbiAgICAgICAgbmV4dDogWydyZXR1cm4nLCAnYmxvY2snLCAnYmxvY2stbGlrZSddXG4gICAgICB9XSAqL1xuY29uc3QgZGlnaXRzID0gW107XG5jb25zdCBkaWdpdCA9IGRpZ2l0cy5wb3AoKTtcblxuaWYgKGRpZ2l0KSB7XG4gIGNvbnNvbGUubG9nKGRpZ2l0KTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

I'm going to close this as it's not a problem with this project. But feel free to discuss more here, or open a new issue in the eslint repo for better visibility and better access to the  maintainers.
